examples/apps/color_finder.py

```diff
@@ -0,0 +1,475 @@
+            in self.pixel_is_surrounded_by_color
+        '''
+        for i in range(1,self.grid_width-1):
+            for j in range(1,self.grid_height-1):
+                color = self.pixel_is_surrounded_by_color(i,j)
+                if color:
+                    self.pixel_matrix[i][j]=color
+
+    def pixel_is_surrounded_by_color(self,i,j):
+        ''' If a pixel is surrounded by three pixels of the same color, returns that color. Otherwise, returns None.'''
+        if self.pixel_matrix[i-1][j]==self.pixel_matrix[i][j-1] and self.pixel_matrix[i-1][j]==self.pixel_matrix[i+1][j]:
+            return self.pixel_matrix[i-1][j]
+        if self.pixel_matrix[i-1][j]==self.pixel_matrix[i][j-1] and self.pixel_matrix[i-1][j]==self.pixel_matrix[i][j+1]:
+            return self.pixel_matrix[i-1][j]
+        if self.pixel_matrix[i-1][j]==self.pixel_matrix[i+1][j] and self.pixel_matrix[i-1][j]==self.pixel_matrix[i][j+1]:
+            return self.pixel_matrix[i-1][j]
+        if self.pixel_matrix[i+1][j]==self.pixel_matrix[i][j-1] and self.pixel_matrix[i+1][j]==self.pixel_matrix[i][j+1]:
+            return self.pixel_matrix[i+1][j]
+        return None 
+
+    async def on_new_camera_image(self, evt, **kwargs):
+        ''' Processes the blobs in Cozmo's view.
+            Triggered by EvtOnNewCameraImage ~15 times a second.
+        '''
+        downsized_image = self.get_low_res_view()
+        self.update_pixel_matrix(downsized_image)
+        blob_detector = BlobDetector(self.pixel_matrix,self.possible_colors_to_find)
+        blob_info = blob_detector.get_center_of_blob_with_color(self.color_to_find)
+        if blob_info:
+            if self.state=='look_around':
+                self.state='found_color'
+                self.look_around_behavior.stop()
+            self.on_finding_a_blob(blob_info)
+        else:
+            self.robot.set_backpack_lights_off()
+            self.state='look_around'
+
+    def get_low_res_view(self):
+        ''' Uses the resize method from the Pillow library to get a low-resolution
+            version of Cozmo's camera view.
+        '''
+        image = self.robot.world.latest_image.raw_image
+        return image.resize((self.grid_width,self.grid_height), resample = Image.LANCZOS)
+
+    def on_finding_a_blob(self,blob_info):
+        ''' Determines whether Cozmo should continue to look at the blob, or drive towards it.'''    
+        x,y = blob_info
+        WM = 10.0*32/self.grid_width  # WM and HM are multipliers that adjust the amount to turn
+        HM = 10.0*24/self.grid_height # based on the current dimensions of self.pixel_matrix.
+        self.robot.set_center_backpack_lights(map_color_to_light[self.color_to_find])
+
+        # 'fov' stands for 'field of view'. This is the angle amount
+        # that Cozmo can see to the edges of his camera view.
+        amount_move_head = radians(self.robot.camera.config.fov_y.radians*(120-y*HM)/240)
```

Comment:
Adding self.fov_x and self.fov_y as properties of the ColorFinder class so that lookup only occurs during __init__